File: Medium/Don't Panic - Episode 1/Don_t_Panic_Episode_1.py
nb_floors, width, nb_rounds, exit_floor, exit_pos, nb_total_clones, nb_additional_elevators, nb_elevators = [int(i) for i in input().split()]
elevators = {}
for i in range(nb_elevators):
    elevator_floor, elevator_pos = [int(j) for j in input().split()]
    elevators[elevator_floor] = elevator_pos
    
while True:
    clone_floor, clone_pos, direction = input().split()
    clone_floor = int(clone_floor)
    clone_pos = int(clone_pos)

    if (clone_floor < 0) or (clone_floor == exit_floor):
        direction2Exit = ("RIGHT", "LEFT")[exit_pos - clone_pos > 0]
        result = ("BLOCK", "WAIT")[direction2Exit != direction]
    else:
        elevator_pos = elevators[clone_floor]
        direction2Elevator = ("RIGHT", "LEFT")[elevator_pos - clone_pos > 0]

        if (elevator_pos - clone_pos == 0):
            result = "WAIT"
        else:
            result = ("BLOCK", "WAIT")[direction2Elevator != direction]

    print(result)

File: Medium/Don't Panic - Episode 1/Don_t_Panic_Episode_1.js
var inputs = readline().split(' ');
const nbFloors = parseInt(inputs[0]);               // number of floors
const width = parseInt(inputs[1]);                  // width of the area
const nbRounds = parseInt(inputs[2]);               // maximum number of rounds
const exitFloor = parseInt(inputs[3]);              // floor on which the exit is found
const exitPos = parseInt(inputs[4]);                // position of the exit on its floor
const nbTotalClones = parseInt(inputs[5]);          // number of generated clones
const nbAdditionalElevators = parseInt(inputs[6]);  // ignore (always zero)
const nbElevators = parseInt(inputs[7]);            // number of elevators
let elevatorFloor = [];
let elevatorPos = [];

for (let i = 0; i < nbElevators; i++) {
    var inputs = readline().split(' ');
    elevatorFloor.push(parseInt(inputs[0]));      // floor on which this elevator is found
    elevatorPos.push(parseInt(inputs[1]));        // position of the elevator on its floor
    
}

// game loop
while (true) {
    var inputs = readline().split(' ');
    const cloneFloor = parseInt(inputs[0]); // floor of the leading clone
    const clonePos = parseInt(inputs[1]);   // position of the leading clone on its floor
    const direction = inputs[2];            // direction of the leading clone: LEFT or RIGHT

    // S'il n'y a pas d'ascenseur
    if (elevatorFloor.length == 0 || elevatorFloor.indexOf(cloneFloor) == -1){SameFloor(elevatorFloor, exitPos, clonePos, direction);}
    // S'il y a un ascenseur, il faut le trouver
    else{SameFloor(elevatorFloor, elevatorPos[elevatorFloor.indexOf(cloneFloor)], clonePos, direction)}
}


function SameFloor(elevatorFloor, exitPos, clonePos, direction){
        let d = exitPos - clonePos;

        if((d<0 && direction == 'RIGHT') || (d>0 && direction == 'LEFT')) {console.log('BLOCK');}
        else{console.log('WAIT');}
}
